### PR TITLE
opt replace-std-json-with-json-iterator

### PR DESCRIPTION
**Pull request by Chyroc**

### CODE REVIEW

**Comment by Chyroc** [comment]
see: https://github.com/json-iterator/go 

and:

 

it is faster ans faster.

**Comment by Chyroc** [comment]
see this : https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go#L26

k8s already used it..

File: decoder.go [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 	"encoding/xml"
 	"errors"
 	"io"
 	"io/ioutil"
 	"net/http"
+
+	"github.com/json-iterator/go"
 )
 
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
```
Comment (VojtechVitek) [inline decoder.go]: why do they require a variable, why not a drop-in import replacement?

Comment (Chyroc) [reply decoder.go]: Perhaps to distinguish it from the standard library

**Comment by Chyroc** [comment]
@VojtechVitek Have a look, I have set him as optional。
and use it：
```go
func use() {
	// use json std
	json.Marshal(map[string]string{"ooo": "json std ..."})

	// use json-iterator
	RegisterJSONIterator()
	json.Marshal(map[string]string{"ooo": "json-iterator ..."})
}
```

**Comment by Chyroc** [comment]
@VojtechVitek Maybe it can be merged, or do you have any further comments? thanks.

**Pull request closed without merging**